### PR TITLE
Core: Warn about jQuery.isNumeric differences

### LINKED ISSUE #134: Restore 2.2 isNumeric signature
jquery/jquery/issues/2662

jQuery 3 changed `jQuery.isNumeric` to only return true on number/string input, whereas it would previously use `toString` and exempt only arrays. Migrate should restore and warn about the old behavior.

### PR DESCRIPTION
**Pull request by dmethvin**
Closes #134 

It just seemed simplest to put the old implementation in the code and compare the two.

Are there any other edge cases we need to test here? Is there a better warning message?

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
 var oldInit = jQuery.fn.init,
+	oldIsNumeric = jQuery.isNumeric,
```
Comment (dmethvin) [inline src/core.js]: Dunno why I did this, I could have called it directly since I'm not patching it. :smile_cat:

Comment (gibson042) [inline src/core.js]: Not at all; LGTM.
